src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -344,26 +344,21 @@ private BoundExpression CreateTupleConversion(CSharpSyntaxNode syntax, BoundTupl
 
             NamedTypeSymbol targetType = (NamedTypeSymbol)destinationWithoutNullable;
 
-            NamedTypeSymbol targetUnderlyingType;
             if (targetType.IsTupleType)
             {
                 var destTupleType = (TupleTypeSymbol)targetType;
-                targetUnderlyingType = destTupleType.UnderlyingTupleType;
                 // do not lose the original element names in the literal if different from names in the target
+
+                // PROTOTYPE(tuples): Come back to this, what about locations?
                 targetType = destTupleType.WithElementNames(sourceTuple.ArgumentNamesOpt);
```

Comment:
That is for cases like
```C#
( (long a, long b) ) ( c: 1, d: 2 );
```
The literal is target-typed to (long c, long d), because target-typing of elements affects only types, not names. I think that was the idea. We can change that.

Then there is a conversion node that identity-converts to (long a, long b). 
The result *is*  (long a, long b)